api/validation.go

```diff
@@ -60,10 +60,20 @@ func (api *API) validateAndHydrateDimensions(v dataset.Version, dims []model.Dim
 	}
 
 	if v.IsBasedOn.Type == cantabularMultivariateTable {
-		resp, err := api.getCantabularDimensions(ctx, dims, pType)
+
+		if postDimension {
```

Comment:
For a cantabularMultivariateTable if the portDimension is 'false' we are not following the process in which dimensions were populated in case of cantabularFlexibleTable https://github.com/ONSdigital/dp-cantabular-filter-flex-api/blob/develop/api/validation.go#L28

Reply:
yes we are I believe... as we then will be hitting this block `
		resp, err := api.getCantabularDimensions(ctx, dims, pType, postDimension)
		if err != nil {
			return nil, errors.Wrap(err, "failed to get dimensions from Cantabular")
		}

		return resp, nil`

Reply:
For filter-flex table type **getCantabularDimensions** is being called for the range of dims, but for multivariates if portDimension is false we are not doing the same. So either cantabularFlexibleTable is over-doing it or cantabularMultivariateTable is under-doing (or I didn't understood it well)
`for _, d := range dims {
			if d.IsAreaType != nil && *d.IsAreaType {
				areaCount++
				geodims, err := api.getCantabularDimensions(ctx, []model.Dimension{d}, pType, false)
				if err != nil {
					return nil, errors.Wrap(err, "failed to get geography dimension from Cantabular")
				}
				geodim = geodims[0]
			}
		}`